### PR TITLE
feat: Delaying paint until Activation

### LINKED ISSUE #11974: [Android] Implement InternalActivate to delay first draw of application until Activate is called
## What would you like to be added:

The Window class has an [InternalActivate](https://github.com/unoplatform/uno/blob/d077eaf8b9337c55c0fd9e912e65f99088a3ab9d/src/Uno.UI/UI/Xaml/Window.cs#L212) partial method which should be implemented by each platform to determine when the first draw of the window should occur. This is not implemented on Android and should be to control when the splash screen is dismissed and the first draw of the application content occurs

## Why is this needed:

Other platforms implement the InternalActivate method on the Window class in order to control the first draw of the application (eg [iOS](https://github.com/unoplatform/uno/blob/d077eaf8b9337c55c0fd9e912e65f99088a3ab9d/src/Uno.UI/UI/Xaml/Window.iOS.cs#L79-L82) and [wasm](https://github.com/unoplatform/uno/blob/d077eaf8b9337c55c0fd9e912e65f99088a3ab9d/src/Uno.UI/UI/Xaml/Window.wasm.cs#L134-L137) ) but Android does not. 

Currently Android will display the first content of the application as soon as it's available to be drawn, rather than waiting for the Activate method to be called. This means that the native splash screen will be dismissed prematurely.

## For which Platform:

- [ ] WebAssembly
- [X] Android
- [ ] iOS
- [ ] macOS (AppKit)
- [ ] Mac Catalyst
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Linux Framebuffer
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by nickrandolph**
GitHub Issue (If applicable): closes #11974 

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the current behavior?

Android application ui displays prior to Window.Activate being called

## What is the new behavior?

Splashscreen stays visible until Window.Activate is called at which point the application ui is shown.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Window.Android.cs [hunk 1]
```diff
@@ -53,6 +63,19 @@ private void InternalSetContent(UIElement value)
...
+			if (!_preDrawListener.IsActivated)
+			{
+				var activity = Uno.UI.ContextHelper.Current as Android.App.Activity;
+				if (activity is not null)
+				{
+					_decor = activity.Window.DecorView;
+					if (_decor is not null)
+					{
+						_decor.ViewTreeObserver.AddOnPreDrawListener(_preDrawListener);
+					}
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Window.Android.cs:76]:
```suggestion
					_decor?.ViewTreeObserver.AddOnPreDrawListener(_preDrawListener);
```

File: src/Uno.UI/UI/Xaml/Window.Android.cs [hunk 2]
```diff
@@ -334,6 +357,23 @@ private bool IsNavigationBarTranslucent()
 				|| flags.HasFlag(WindowManagerFlags.LayoutNoLimits);
 		}
 		#endregion
+
+
+		private class ActivationPreDrawListener : Java.Lang.Object, ViewTreeObserver.IOnPreDrawListener
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Window.Android.cs:362]:
```suggestion
		private sealed class ActivationPreDrawListener : Java.Lang.Object, ViewTreeObserver.IOnPreDrawListener
```

**Review by jeromelaban — APPROVED** [review]
🚀

**Comment by MartinZikmund** [comment]
@nickrandolph @jeromelaban  sorry, I didn't notice this earlier... I have a PR open for this for a while now, it waited for review - https://github.com/unoplatform/uno/pull/11864 - but handles a bit more alignment with WinUI / UWP, so it is still relevant - please take a look and I will rebase on this work

**Comment by nickrandolph** [comment]
@MartinZikmund did you need to modify your pr to remove/change this implementation?

**Comment by MartinZikmund** [comment]
> @MartinZikmund did you need to modify your pr to remove/change this implementation?

I haven't rebased it yet - but I think the main difference is that my PRs delay is waiting for Visible as compared to Activated (as Visible happens only after activate, and in WinUI the splash is still on in between those two events). But the crux of the logic is the same, so it should not be a breaking change

**Comment by FrozDark** [comment]
Is it a regression?

https://user-images.githubusercontent.com/6670150/235157795-e933fe76-64a8-4b44-89ee-f1716e85a9d9.mp4

**Comment by jeromelaban** [comment]
@FrozDark which build are you using?

/cc @nickrandolph

**Comment by FrozDark** [comment]
> @FrozDark which build are you using?
> 
> /cc @nickrandolph
 Latest dev 739

**Comment by nickrandolph** [comment]
It's likely that Activate is being called late in your startup sequence. Try moving Activate to earlier in startup.

**Comment by FrozDark** [comment]
> It's likely that Activate is being called late in your startup sequence. Try moving Activate to earlier in startup.

That worked
But what is that a white background flickering between page transitions as on video? It still persists

**Comment by nickrandolph** [comment]
@FrozDark are you using either Uno.Extensions navigation or ExtendedSplashScreen from toolkit? 

Are you setting the Background colour explicitly or using a ThemeResource

Is your device set to Light or Dark theme?

Can I ask you to raise a separate issue with more information so we can track the flicker as I don't think it's related to the changes in activation

**Comment by jeromelaban** [comment]
@FrozDark would you mind opening an issue about the flickering behavior? Discussing in a PR is difficult to follow :)

**Comment by MartinZikmund** [comment]
@FrozDark did you open an issue regarding this please?

**Comment by FrozDark** [comment]
@nickrandolph @jeromelaban @MartinZikmund 
My bad didn't informed you here
It's fixed by this PR https://github.com/unoplatform/uno/pull/11864
No issue now

**Pull request merged**